torch/distributed/_tensor/__init__.py

```diff
@@ -20,6 +20,97 @@
 ]
 
 
+def _dtensor_init_helper(init_op, *size, **kwargs) -> DTensor:
```

Comment:
how about we don't use `**kwargs` and directly use the sth like:
```
_dtensor_init_common(init_op, *size, requires_grad=.., dtype=..., layout=..
```
So that we don't really need those `kwargs.pop` and be explicit about what the default args are.

Reply:
The reason why I use `kwargs` is for some ops have more keyword-only arguments: for example, `torch.empty` has `pin_memory=False, memory_format=torch.contiguous_format` but `ones` and `zeros` don't have these.

Reply:
I see, got it. My feeling: we don't really need to conform exactly with those tensor constructor differences. i.e. we already not exposing `device` arg, I feel for `empty`, we should probably not expose `pin_memory` and `memory_format`. My feeling is that these two options are rarely used nowadays so we don't need to special handle them.

Reply:
yeah that simplifies the whole thing if we make the argument list uniform over init constructors. I'll rewrite them in next update. I'm also drafting `torch.full` support and I find re-using `_dtensor_init_common` for `full` is not straightforward. Either make it more generic or not using it when implementing `full`. I need to think about it further...

Reply:
I see, Is the issue coming from `fill_value`?

Here is a more concrete suggestion, `_dtensor_init_common` could have the following signature:

`(init_op, size: torch.Size, fill_value = None, device_mesh=None, placements=None,  **kwargs)`

For each constructor, we prepare the `size` first before passing in to the common method, and pass the rest as kwargs, for `torch.fill`, we probably do a check and pass in `size, fill_value` for that case. `kwargs` can be directly passed in to the `init_op`.

Reply:
my worry is we may need to keep expanding the keyword-only argument list when we keep adding more init constructors. For example, to add `torch.arange` op, we will need to add `start`, `end, `step` to the list and it seems not a good approach.

One proposal is we define it as `_dtensor_init_common(init_op, size: torch.Size, device_mesh=None, placements=None, **kwargs)` and we check `size` and `kwargs` to see of which "type" `init_op` is. 

e.g. `size` is not None, we check if `kwargs` has `fill_value`. If yes, call `init_op` with positional arguments `size` and `fill_value`. If no,  call `init_op` only with positional argument `size`. If `size` is None, then the op may be `arange` so we check if `kwargs` has `start` in it. If yes, pass it as positional argument. If no, pass `kwargs` to init_op which would eventually throw an error. 

What do you think?

Reply:
Yeah that works too. 

For checking logic, we can just simply check if `init_op == full` or `init_op == arange` and handle things correspondingly, that should make things simpler